### PR TITLE
14546 Added editable business name

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14546

*Description of changes:*
- added editable business name

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/business-lookup/BusinessLookup.vue [hunk 1]
```diff
@@ -41,27 +41,45 @@
 
     <div v-if="state === States.SUMMARY && haveBusiness" class="summary-block mt-5">
       <v-row no-gutters>
-        <v-col cols="9">
+        <v-col cols="10">
```
Comment (severinbeauvais) [inline src/components/business-lookup/BusinessLookup.vue:44]: Example:
image

File: src/components/business-lookup/BusinessLookup.vue [hunk 2]
```diff
@@ -41,27 +41,45 @@
...
           <v-row no-gutters>
             <v-col cols="12">
-              <label>Name: </label>
-              <span>{{businessName}}</span>
+              <div v-if="editableBusinessName" class="d-flex align-center">
```
Comment (severinbeauvais) [inline src/components/business-lookup/BusinessLookup.vue:47]: Class ref: https://vuetifyjs.com/en/styles/flex/

Example:
image

File: src/components/business-lookup/BusinessLookup.vue [hunk 3]
```diff
@@ -41,27 +41,45 @@
...
+                <v-text-field
+                  dense
+                  filled
+                  hide-details="auto"
+                  id="organization-name"
+                  class="mx-4 mr-md-0"
+                  :rules="businessNameRules"
+                  :value="businessName"
+                  @input="setBusinessName($event)"
+                />
```
Comment (severinbeauvais) [inline src/components/business-lookup/BusinessLookup.vue:58]: Component ref: https://vuetifyjs.com/en/api/v-text-field/

File: src/components/business-lookup/BusinessLookup.vue [hunk 4]
```diff
@@ -111,11 +132,17 @@ export default class BusinessLookup extends Vue {
...
+  /** The business lookup validation rules. */
   readonly businessLookupRules: Array<(v) => boolean | string> = [
     v => !!v || 'Business is required'
   ]
 
+  /** The business name validation rules. */
+  readonly businessNameRules: Array<(v) => boolean | string> = [
+    (v: string) => !!v?.trim() || 'Business or corporation name is required',
+    (v: string) => (v?.length <= 150) || 'Cannot exceed 150 characters' // maximum character count
+  ]
```
Comment (severinbeauvais) [inline src/components/business-lookup/BusinessLookup.vue:144]: Same rules as in Edit UI.

Ref: https://github.com/bcgov/business-edit-ui/blob/a3a1416a82d14de2c299e21f04ec18e8ca4a5a1b/src/components/common/PeopleAndRoles/OrgPerson.vue#L835

**Review by ketaki-deodhar — APPROVED** [review]
Looks good

**Review by ChrisGabel23 — APPROVED** [review]

**Pull request merged**